File: src/Backend/Sudoku/Application/Datasets/PreparedDatasetSourceReportDto.cs
namespace Sudoku.Application.Datasets;

public sealed record PreparedDatasetSourceReportDto(
    string Name,
    string RequestedType,
    string DetectedType,
    int ProcessedSampleCount,
    int IncludedSampleCount,
    int EmptyCellCount,
    int RejectedSampleCount,
    IReadOnlyList<string> Warnings);

File: src/MachineLearning/application/features/datasets/dto/prepared_dataset_source_report_dto.py
from dataclasses import dataclass


@dataclass(frozen=True)
class PreparedDatasetSourceReportDto:
    name: str
    requested_type: str
    detected_type: str
    processed_sample_count: int
    included_sample_count: int
    empty_cell_count: int
    rejected_sample_count: int
    warnings: tuple[str, ...]
